File: ABC/351-400/385/b.py
def main():
    h,w,x,y = map(int,input().split())
    s = [input() for _ in range(h)]
    t = input()
    visited = [[False]*w for _ in range(h)]
    x -= 1
    y -= 1
    visited[x][y] = True
    for c in t:
        nx,ny = x,y
        if c == 'U':
            nx -= 1
        elif c == 'D':
            nx += 1
        elif c == 'L':
            ny -= 1
        elif c == 'R':
            ny += 1
        if nx < 0 or ny < 0 or nx >= h or ny >= w:
            continue
        if s[nx][ny] == '#':
            continue
        visited[nx][ny] = True
        x = nx; y = ny
    
    ans = 0
    for i in range(h):
        for j in range(w):
            if visited[i][j] and s[i][j] == '@':
                ans += 1
    
    print(x+1,y+1,ans)

if __name__ == "__main__":
    main()

File: ABC/351-400/385/b.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<n;++i)

int main(){
    int h,w,si,sj;
    cin >> h >> w >> si >> sj;
    si--; sj--;
    vector<string> s(h);
    rep(i,h) cin >> s[i];
    string t;
    cin >> t;
    vector<vector<bool> > visited(h,vector<bool>(w));
    visited[si][sj] = true;
    int ci = si; int cj = sj;
    map<char,int> mp;
    mp['U'] = 0;
    mp['D'] = 1;
    mp['L'] = 2;
    mp['R'] = 3;
    int di[] = {-1,1,0,0};
    int dj[] = {0,0,-1,1};
    rep(k,t.size()){
        char c = t[k];
        int ni = ci + di[mp[c]];
        int nj = cj + dj[mp[c]];
        if(ni < 0 || nj < 0 || ni >= h || nj >= w) continue;
        if(s[ni][nj] == '#') continue;
        visited[ni][nj] = true;
        ci = ni; cj = nj;
    }
    int ans = 0;
    rep(i,h)rep(j,w){
        if(s[i][j] == '@' && visited[i][j]) ans++;
    }
    cout << ci+1 << ' ' << cj+1 <<  ' ' << ans << endl;
    return 0;
}